modules/lib/types-dag.nix

```diff
@@ -65,7 +65,21 @@ rec {
     mkOptionType rec {
       name = "dagOf";
       description = "DAG of ${elemType.description}";
-      inherit (attrEquivalent) check merge emptyValue;
+      check = {
+        __functor = _self: attrEquivalent.check;
+        isV2MergeCoherent = true;
+      };
+      merge = {
+        __functor =
+          self: loc: defs:
+          (self.v2 { inherit loc defs; }).value;
+        v2 =
+          { loc, defs }:
+          attrEquivalent.merge.v2 {
+            inherit loc defs;
+          };
```

Comment:
IIRC, there's a `lib` function for calling either v1 or v2 merge (whichever is available on a given type), which provides some backwards compatibility.

IIUC, that's not strictly needed in this case, unless it is possible the attrsOf can come from a different revision of nixpkgs-lib. Otherwise it should be safe to assume there is a `attrsEquivialent.merge.v2`.